src/parser/serializer.ts

```diff
@@ -757,6 +756,8 @@ export function serializeInto(
         index = serializeDate(buffer, key, value, index);
       } else if (value === null || (value === undefined && ignoreUndefined === false)) {
         index = serializeNull(buffer, key, value, index);
+      } else if (value._bsontype === 'ObjectId') {
+        index = serializeObjectId(buffer, key, value, index);
```

Comment:
I did see a measurable performance gain by moving the ObjectId check to the top `if` in my testing, I believe since some of those checks like `isUInt8Array()` and `Symbol.for("@@mdb.bson.version")` are moderately expensive for such a hot path. Now that I'm thinking about this we can instantiate a single Symbol variable and just reference it instead of creating it every time, and maybe move the `isUInt8Array()` down lower if that doesn't break any tests.